### ISSUE TITLE
Update documentation to address situation where wildcard source path is accompanied by other source paths

### ISSUE THREAD

**Issue by eecavanna** [documatt/sphinx-reredirects#9]
Thanks for building this plugin!

Today, I tried implementing two redirects on a site. One of the source paths had a wildcard in it, and the other one did not. Here's a representative example:

```py
redirects = {
   "path/to/foo" = "https://www.example.com/foo",
   "path/to/*" = "https://www.example.com/other"
}
```

I had expected the outcome to be:
- If someone visits `path/to/foo.html`, they would be redirected to `https://www.example.com/foo`
- If someone visits `path/to/bar.html`, they would be redirected to `https://www.example.com/other`

What I observed was:
- If someone visits `path/to/foo.html`, they are redirected to `https://www.example.com/other`
- If someone visits `path/to/bar.html`, they are redirected to `https://www.example.com/other`

In other words, the source path containing the wildcard seemed to me to take priority.

I'm opening this Issue to request that the behavior be addressed in the [plugin's documentation](https://documatt.com/sphinx-reredirects/usage.html#wildcard-syntax). Currently, in all the examples where a source path includes a wildcard, there is only one redirect in that example.

**Comment by eecavanna** [comment]
> Currently, in all the examples where a source path includes a wildcard, there is only one redirect in that example.

This is true about all examples, regardless of whether a source path includes a wildcard. All examples have only one redirect defined in the `redirects` dictionary.

**Comment by pmatulis** [comment]
The more specific case should take precedence surely. If not, that's a bug.
